services/venice-router/src/main/java/com/linkedin/venice/router/MetaDataHandler.java

```diff
@@ -316,6 +316,14 @@ private void handleValueSchemaLookup(ChannelHandlerContext ctx, VenicePathParser
       int index = 0;
       for (SchemaEntry entry: valueSchemaEntries) {
         int schemaId = entry.getId();
+        if (schemaId < 1) {
```

Comment:
If that ever happened then the array setting below would fail with `IndexOutOfBoundsException`... I wonder if that ever happened, but it doesn't hurt to have the defensive code!

Reply:
Good catch! added some tests as well.